src/interactive/azext_interactive/azclishell/az_completer.py

```diff
@@ -259,21 +281,57 @@ def gen_dynamic_completions(self, text):
         except Exception:  # pylint: disable=broad-except
             pass
 
+    def gen_recommend_completion(self, text):
+        """ generates recommended commands """
+        recommend_result = self.shell_ctx.recommender.get_commands() or []
+        if not recommend_result:
+            default_recommendations = self.shell_ctx.recommender.get_default_recommendations()
+            for rec in default_recommendations:
+                description = default_recommendations[rec]
+                description = description or self.command_description.get(rec, '')
+                if text.strip() == '':
+                    yield Completion(rec, 0, display_meta=description)
+        for rec in recommend_result:
+            description = self.command_description.get(rec['command'], 'Commonly used command by other users')
+            if text == '':
+                yield Completion(rec['command'], 0, display_meta=description)
+            elif text == 'a':
+                yield Completion('az ' + rec['command'], -1, display_meta=description)
+            else:
+                formatted_text = re.sub(r'\s+', ' ', text).strip()
+                if rec['command'].startswith(formatted_text) and rec['command'] != formatted_text:
+                    yield Completion(rec['command'], -len(text.lstrip()), display_meta=description)
+
+    def gen_recommended_scenario(self, text):
+        """ generates recommended scenarios """
+        if '-' in text or not re.fullmatch(r'[a-zA-Z\s]+', text):
+            return
+        recommend_result = self.shell_ctx.recommender.get_scenarios() or []
+        for idx, rec in enumerate(recommend_result):
+            yield Completion(
+                '::' + str(idx + 1), -len(text),
+                display_meta=f'{rec["scenario"]} ({len(rec["nextCommandSet"])} Commands)',
+                display=f'* command set {str(idx + 1)}')
+
     def yield_param_completion(self, param, last_word):
         """ yields a parameter """
         return Completion(param, -len(last_word), display_meta=self.param_description.get(
             self.current_command + " " + str(param), '').replace(os.linesep, ''))
 
     def gen_cmd_and_param_completions(self):
         """ generates command and parameter completions """
+        if not self.current_command and not self.unfinished_word.strip() and self.shell_ctx.recommender.enabled:
+            return
```

Comment:
Please add a comment to explain this logic

Reply:
Done